packages/SwingSet/tools/run-utils.ts

```diff
@@ -0,0 +1,152 @@
+  // queued messages themselves.  The status and resolution values of these
+  // promises can be synchronously queried (by kpid) via the
+  // `controller.kpStatus` and `controller.kpResolution` methods once `run` has
+  // completed.  These queries are only available once the swingset has
+  // reqlinquished agency, i.e., when the work initiated by `controller.run` has
+  // finished.  At that point, nothing is going on inside the kernel, and
+  // nothing WILL be going on inside the kernel until a subsequent call to
+  // `controller.run`, which in turn will only have an effect if additional
+  // messages have been placed on the kernel run queue in the meantime.  You MAY
+  // NOT call `queueToVatRoot`, `queueToVatObject`, `kpStatus`, or
+  // `kpResolution` while run queue execution, triggered by a call to `run`, is
+  // in progress
+  //
+  // The functionality made available by `EV` looks similar to that provided by
+  // `E`, but it is very much not.  When you send a message using `EV`, it
+  // places the message onto the kernal run queue and then immediately invokes
+  // `controller.run`.  When the result of `run` resolves, the kpid returned by
+  // the message enqueueing operation is queried.  If at that time the promise
+  // it identifies is resolved (or rejected), the value it was resolved (or
+  // rejected) to is used as the result from the `EV` invocation.  However, if
+  // it is still pending at that time, `EV` will throw an exception and your
+  // test will abort.  This means that if you initiate some operation via an
+  // `EV` message send, it must complete within a single `run` cycle for it to
+  // be of any use to you.  This is quite different from a message sent using
+  // `E`, which will return a promise that can remain pending indefinitely,
+  // possibly to be resolved by a future message delivery.
+
+  // @ts-expect-error cast, approximate
+  const EV: EVProxy = presence =>
+    new Proxy(harden({}), {
+      get: (_t, method, _rx) =>
+        harden((...args) =>
+          queueAndRun(() =>
+            controller.queueToVatObject(presence, method, args),
+          ),
+        ),
+    });
+  EV.vat = vatName =>
+    new Proxy(harden({}), {
+      get: (_t, method, _rx) =>
+        harden((...args) =>
+          queueAndRun(() => controller.queueToVatRoot(vatName, method, args)),
+        ),
+    });
+  // @ts-expect-error xxx
+  EV.sendOnly = presence =>
+    new Proxy(harden({}), {
+      get: (_t, method, _rx) =>
+        harden((...args) =>
+          queueAndRun(
+            () => controller.queueToVatObject(presence, method, args),
+            true,
+          ),
+        ),
+    });
+  // @ts-expect-error xxx
+  EV.get = presence =>
+    new Proxy(harden({}), {
+      get: (_t, pathElement, _rx) =>
+        queueAndRun(() =>
+          controller.queueToVatRoot('bootstrap', 'awaitVatObject', [
+            presence,
+            [pathElement],
+          ]),
+        ),
+    });
```

Comment:
This assumption of a bootstrap-vat `awaitVatObject` method is giving off some bad vibes; is there a way to do it with just the controller? Untested idea:
```suggestion
  /**
   * Derives a value suitable for immediate interaction from a vat-originating value:
   * the fulfillment value of a fulfilled promise, or
   * the rejection value of a rejected promise, or
   * a non-promise value itself.
   * If `value` is an unsettled promise, a synchronous error will be thrown.
   *
   * @param {any} presence
   * @returns {{status: 'fulfilled' | 'rejected' | undefined, value: any}}
   */
  const presenceSettlement = (presence, errPrefix = '') => {
    if (passStyleOf(presence) === 'promise') {
      const kpid = krefOf(presence);
      const status = controller.kpStatus(kpid);
      switch (status) {
        case 'fulfilled':
          // falls through
        case 'rejected':
          return { status, value: kunser(controller.kpResolution(kpid)) };
        case 'unresolved':
          throw Fail(`${b(errPrefix)}unsettled value for ${q(kpid)}`;
        default:
          throw Fail`${b(errPrefix)}unknown promise status ${q(kpid)} ${q(status)}`;
      }
    }
    return { status: undefined, value: presence };
  };
  
  EV.get = presence =>
    new Proxy(harden({}), {
      get: async (_t, key, _rx) => {
        await queueAndRun(() => {}, true);
        const errPrefix = '[EV.get] ';
        const { status, value: obj } = presenceSettlement(presence, errPrefix);
        if (status === 'rejected') {
          throw obj;
        } else if (passStyleOf(obj) === 'remotable') {
          throw Fail`${b(errPrefix)}Cannot read properties from remotable ${q(krefOf(obj))}`);
        }
        const { status: propStatus, value } = presenceSettlement(obj[key], errPrefix);
        if (propStatus === 'rejected') {
          throw value;
        }
        return value;
      },
    });
```